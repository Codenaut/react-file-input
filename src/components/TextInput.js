import React from 'react';
import PropTypes from 'prop-types';

import '../styles/TextField.scss';

const TextInput = ({ placeholder, value, onChange }) => (
  <input
    className="brainhub-text-field__input"
    placeholder={placeholder}
    type="text"
    value={value}
    onChange={onChange}
  />
);

TextInput.defaultProps = {
  value: '',
};

TextInput.propTypes = {
  placeholder: PropTypes.string.isRequired,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

export default TextInput;
