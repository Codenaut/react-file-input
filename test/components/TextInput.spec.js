import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import TextInput from '../../src/components/TextInput';

const defaultProps = {
  placeholder: 'Test Placeholder',
  onChange: () => null,
};

const setup = (props = {}) => {
  const textInput = shallow(<TextInput {...defaultProps} {...props} />);

  return {
    textInput,
    input: textInput.find('input'),
  };
};

describe('components', () => {
  describe('TextInput', () => {
    it('should render a text input', () => {
      const { input } = setup();

      expect(input).toHaveLength(1);
      expect(input.prop('placeholder')).toBe(defaultProps.placeholder);
      expect(input.prop('type')).toBe('text');
      expect(input.prop('value')).toBe('');
    });

    it('should render a text input with the default value provided in props', () => {
      const value = 'test value';
      const { input } = setup({ value });

      expect(input.prop('value')).toBe(value);
    });

    it('should call onChange callback when input changes', () => {
      const onChange = jest.fn();
      const { input } = setup({ onChange });

      input.simulate('change');
      expect(onChange).toHaveBeenCalled();
    });

    it('should match exact snapshot', () => {
      const tree = renderer.create(
        <div>
          <TextInput {...defaultProps} />
        </div>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
