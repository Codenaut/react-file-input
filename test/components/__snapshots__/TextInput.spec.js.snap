// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components TextInput should match exact snapshot 1`] = `
<div>
  <input
    className="brainhub-text-field__input"
    onChange={[Function]}
    placeholder="Test Placeholder"
    type="text"
    value=""
  />
</div>
`;
